newm-server/src/main/kotlin/io/newm/server/features/minting/MintingMessageReceiver.kt

```diff
@@ -39,6 +40,13 @@ class MintingMessageReceiver : SqsMessageReceiver {
     override suspend fun onMessageReceived(message: Message) {
         val mintingStatusSqsMessage: MintingStatusSqsMessage = json.decodeFromString(message.body)
         log.info { "received: $mintingStatusSqsMessage" }
+        val dbSong = songRepository.get(mintingStatusSqsMessage.songId)
+        if (dbSong.mintingStatus != mintingStatusSqsMessage.mintingStatus) {
+            // sometimes, we will send an SQS message manually to re-process a given song. This is fine, but we need to
+            // sync the DB with the SQS message.
+            log.warn { "DB MintingStatus: ${dbSong.mintingStatus} does not match SQS MintingStatus: ${mintingStatusSqsMessage.mintingStatus}... updating db" }
+            songRepository.update(mintingStatusSqsMessage.songId, Song(mintingStatus = mintingStatusSqsMessage.mintingStatus))
+        }
```

Comment:
Instead of `.update()` we can use `.updateSongMintingStatus()` here.

Reply:
`updateSongMintingStatus()` will cause an extra SQS message we don't want.